PROCESSES.md

```diff
@@ -0,0 +1,147 @@
+files) is one of:
+
+- The mostly recently released package version
+- The version that's *about* to be released (see below)
+- A prerelease with a suffix of 00 to indicate that the *next* release
+  should be a first prerelease, typically of a new minor version,
+  due to changes that have been merged. This is to avoid accidentally
+  including new features in a patch release. For example, having
+  released Google.Cloud.Storage.V1 version 2.0.0, new features were
+  merged into the `master` branch and the version was immediately
+  changed to 2.1.0-alpha00
+
+Typically version numbers should be changed in a commit which does
+nothing else, for clarity. Include both the `apis.json` change and
+the project file changes that occur when the project generator has
+been run, in the same commit.
+
+Dependencies in the API catalog are specified as properties where
+the property name is the package name and the value is the version
+number. If the version number is left blank, then:
+
+- For GAX or gRPC dependencies, the version number is determined by
+  the project generator, to keep these dependencies in sync for all
+  packages.
+- For projects within the repository, a project reference is used.
+
+If a version number is specified, it's *always* a package reference.
+Importantly, project references should only be used between
+production packages when both packages will be published together.
+This avoids inconsistent builds where a package is built against a
+local version which is different from the declared dependency. *This
+isn't currently checked, but will be as part of the releasing
+process.*
+
+## Releasing
+
+Releasing consists of five steps:
+
+- Updating the version number in GitHub
+- Tagging the commit (typically the commit that updated the version
+  number)
+- Building and testing
+- Pushing the package to nuget.org
+- Updating the documentation in GitHub (in the `gh-pages` branch)
+
+### Current process
+
+- After the version number is committed, run `tagreleases.sh` from the
+  root directory to tag all updated versions.
+- Run `buildrelease.sh` manually, specifying one of the API names
+  and versions just tagged.
+- Go into the `releasebuild/nuget` directory and push
+- Make sure you have a clone of the gh-pages branch, copy the
+  documentation, commit and push.
+  
+Issues with this:
+
+- Integration tests are not run automatically
+- The build takes a long time as it rebuilds all packages, runs unit
+  tests for them all and builds all the documentation
+- There's no validation of project dependencies as described above
+
+### Future process
+
+- `tagreleases.sh` will perform validation of project references for
+  packages being tagged
+- A Google build machine (with appropriate secrets) will listen for
+  tags being created
+- When a set of tags is created (with some delay to avoid "spotting"
+  one tag when more are on the way), a build will be run of just the
```

Comment:
The tag commit itself? Do you mean just in case someone had (in parallel) spotted a problem and retagged? Could do... I'm not sure how much benefit there is, but it wouldn't hurt. Added in as a possibility.